### PR TITLE
Reduce the number of choices in onboarding flow [backport]

### PR DESCRIPTION
**Pull request by npretto**
Manual backport of https://github.com/metabase/metabase/issues/46741, closes https://github.com/metabase/metabase/issues/46741

I didn't realize it was supposed to go out on v50 but we do: https://metaboat.slack.com/archives/C063Q3F1HPF/p1724236439696599?thread_ts=1724230345.613399&cid=C063Q3F1HPF

The backports includes all PRs from the epic, outside of the test refactor that was already backported.

There were some conflicts because we changed how we handle colors and that was not backported, so please double check if that's right

### CODE REVIEW

File: frontend/src/metabase/setup/components/CompletedStep/CompletedStep.styled.tsx [hunk 1]
```diff
@@ -5,25 +5,27 @@ import { color } from "metabase/lib/colors";
...
   flex-direction: column;
-  align-items: center;
-  padding: 2rem;
-  margin-bottom: 1.75rem;
+  padding: 2.5rem;
   border: 1px solid ${color("border")};
   border-radius: 0.5rem;
   background-color: ${color("white")};
+  gap: 32px;
+  margin-bottom: 1.75rem;
```
Comment (WiNloSt) [inline frontend/src/metabase/setup/components/CompletedStep/CompletedStep.styled.tsx:13]: Why do we remove `margin-bottom` [on master](https://github.com/metabase/metabase/pull/46783/files#diff-32fa510a1d23f2fe49fd596fc072a20377e836c9d804c3f3b51776719747de2e) but leave it in the backport?

**Review by WiNloSt — COMMENTED** [review]
There's only 1 difference from the 3 PRs
- [make the newsletter signup a checkbox instead of a button](https://github.com/metabase/metabase/pull/46783/files#top)
- [skip analytics opt out step for people on cloud](https://github.com/metabase/metabase/pull/46965/files#top)
- [bump onboarding version to 1.3.0](https://github.com/metabase/metabase/pull/47055#top)

Other than that, it's looking good to me.

File: frontend/src/metabase/setup/components/CompletedStep/CompletedStep.styled.tsx (hunk 1 above)
Comment (npretto) [reply frontend/src/metabase/setup/components/CompletedStep/CompletedStep.styled.tsx:13]: That was a bug fixed with https://github.com/metabase/metabase/pull/47021/files which is part of this backport

**Review by WiNloSt — APPROVED** [review]

**Pull request merged**